include/graaflib/algorithm/coloring/welsh_powell.tpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/welsh_powell.h>
+
+#include <unordered_map>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> welsh_powell_coloring(const GRAPH& graph) {
+    
+    using degree_vertex_pair = std::pair<int, vertex_id_t>;
+
+    // Step 1: Sort vertices by degree in descending order
+    std::vector<degree_vertex_pair> degree_vertex_pairs;
+    for (const auto& vertex : graph.get_vertices()) {
+        int degree = graph.get_degree(vertex);
+        degree_vertex_pairs.emplace_back(degree, vertex);
+    }
```

Comment:
Hi @GouraV-Chatterjee, it seems like the CI is still failing to compile. I missed this in the first review, but the `graph` class does not have a method `get_degree`. Rather, it is a free function defined in `properties/vertex_properties.h`.

So once you `#include <graaflib/properties/vertex_properties.h>` you can:

```c++
int degree = properties::vertex_degree(graph, vertex_id);
```

I now also notice that you iterate over the return value of `graph.get_vertices`. However, since this is an unordered map of vertex ID to vertex value, we are iterating over a pair here. i.e.

```suggestion
    for (const auto& [vertex_id, _] : graph.get_vertices()) {
        int degree = properties::vertex_degree(graph, vertex_id);
        degree_vertex_pairs.emplace_back(degree, vertex);
    }
```

Reply:
Got it...I'll make the required changes and notify you!!

Reply:
Thanks! Let me know if you have any issues 👍🏻